### DISCUSSION TITLE
Why SQL doesn't show column option for attachment?

### DISCUSSION THREAD

**Discussion by syeillah** [Budibase/budibase#4570] [Help] [answered]
Hi, i just wondering why "attachment" type when we create new data column were only available on internal budibase DB? 
Because i want to create attachment column in SQL, but there was no "attachment" option. 

<img width="819" alt="Screen Shot 2022-02-19 at 00 05 21" src="https://user-images.githubusercontent.com/99949174/154731197-497aad0c-7c6f-4dea-bcb9-4f396adac8f5.png">

Is there a solution so that I can still create attachment forms by using external data?

**Comment by mjashanks** [comment, marked as answer]
Yea - we just don't support attachment columns in external sources.

You could potentially create a mirrored internal table, with an `externalID` column, and an attachment column - then display both in the design section.

**Reply by xnohat** [reply]
I suggest use S3 File Upload element for external sources

**Reply by syeillah** [reply]
ah okay thankss

**Reply by daawaan4x** [reply]
Hello @mjashanks, are there any plans to support attachment columns in external sources (or atleast some of them)? Since some of the datasources have native JSON column types and JSON can be stored in external sources as text in case it isn't supported anyway
